lib/bap/bap_self.ml

```diff
@@ -60,4 +60,66 @@ module Create() = struct
   let warning f = message Warning ~section:name f
   let error f = message Error ~section:name f
 
+  module Config = struct
+    let plugin_name = name
+    include Bap_config
+
+    let (/) = Filename.concat
+    let confdir = confdir / plugin_name
+    let conf_filename = confdir / "config"
+
+    let get_env_options () =
```

Comment:
This function does _not_ work like `Sys.getenv`. It is more similar to `options ()` but reads from env. Please see my comments below.